File: PlantHere/PlantHere.Application/CQRS/Basket/Queries/GetBasketByUserId/GetBasketByUserIdQueryResult.cs
using PlantHere.Application.CQRS.BasketItem.Queries.GetBasketItems;


namespace PlantHere.Application.CQRS.Basket.Queries.GetBasketByUserId
{
    public class GetBasketByUserIdQueryResult
    {
        public string UserId { get; set; }

        public DateTime CreatedDate { get; set; }

        public List<GetBasketItemsQueryResult> BasketItems { get; set; }

        public decimal TotalPrice { get; set; }

        public decimal DiscountedTotalPrice { get; set; }

    }
}

File: PlantHereNodeJs/Core/PlantHere.Aplication/RequestResponseModels/Basket/Queries/GetBasketByUserId/GetBasketByUserIdQueryResult.js
class GetBasketByUserIdQueryResult {
    constructor(createdDate,userId,basketItems) {
        this.userId = userId,
        this.createdDate = createdDate,
        this.basketItems = basketItems
        this.totalPrice = basketItems.reduce((acc, o) => acc + o.price * o.count, 0)
        this.discountedTotalPrice = basketItems.reduce((acc, o) => acc + o.discountedPrice * o.count, 0)
    }
}

module.exports = { GetBasketByUserIdQueryResult } 